consumer/batching.go

```diff
@@ -306,3 +307,16 @@ func KeysToDelete(entries []*lib.StateChangeEntry) [][]byte {
 	}
 	return keysToDelete
 }
+
+func FilterKeysByPrefix(keys [][]byte, prefix []byte) [][]byte {
```

Comment:
Function looks great - though it doesn't really have a lot to do with the batching process - can we move this to the helpers file?